test/integration/scheduler/scheduler_test.go

```diff
@@ -1314,6 +1316,76 @@ var _ = ginkgo.Describe("Scheduler", func() {
 			util.ExpectPendingWorkloadsMetric(strictFIFOClusterQ, 0, 1)
 		})
+
+		ginkgo.It("Pending workload with StrictFIFO doesn't block other CQ from borrowing from a third CQ", func() {
+			ginkgo.By("Creating ClusterQueues and LocalQueues")
+			cqTeamA := testing.MakeClusterQueue("team-a").
+				ResourceGroup(*testing.MakeFlavorQuotas(onDemandFlavor.Name).Resource(corev1.ResourceCPU, "5").Obj()).
+				Cohort(chName).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, cqTeamA)).Should(gomega.Succeed())
+
+			strictFIFOLocalQueue := testing.MakeLocalQueue("strict-fifo-q", matchingNS.Name).ClusterQueue(strictFIFOClusterQ.Name).Obj()
+			lqTeamA := testing.MakeLocalQueue("team-a-lq", matchingNS.Name).ClusterQueue(cqTeamA.Name).Obj()
+
+			gomega.Expect(k8sClient.Create(ctx, strictFIFOLocalQueue)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, lqTeamA)).Should(gomega.Succeed())
+
+			cqSharedResources := testing.MakeClusterQueue("shared-resources").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas(onDemandFlavor.Name).Resource(corev1.ResourceCPU, "5").Obj()).
+				Cohort(chName).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, cqSharedResources)).Should(gomega.Succeed())
+
+			ginkgo.By("Creating workloads")
+			admittedWl1 := testing.MakeWorkload("wl", matchingNS.Name).Queue(strictFIFOLocalQueue.
+				Name).Request(corev1.ResourceCPU, "3").Priority(10).Obj()
+			gomega.Expect(k8sClient.Create(ctx, admittedWl1)).Should(gomega.Succeed())
+
+			admittedWl2 := testing.MakeWorkload("player1-a", matchingNS.Name).Queue(lqTeamA.
+				Name).Request(corev1.ResourceCPU, "5").Priority(1).Obj()
+			gomega.Expect(k8sClient.Create(ctx, admittedWl2)).Should(gomega.Succeed())
+
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, admittedWl1, admittedWl2)
+			// cqTeamA.Spec.StopPolicy = ptr.To(kueue.Hold)
+			gomega.Eventually(func() error {
+				var cq kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(cqTeamA), &cq)).To(gomega.Succeed())
+				cq.Spec.StopPolicy = ptr.To(kueue.Hold)
+				return k8sClient.Update(ctx, &cq)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			//pendingWl exceed nominal+borrowing quota and could not preempt due to low priority.
+			pendingWl := testing.MakeWorkload("pending-wl", matchingNS.Name).Queue(strictFIFOLocalQueue.
+				Name).Request(corev1.ResourceCPU, "3").Priority(9).Obj()
+			gomega.Expect(k8sClient.Create(ctx, pendingWl)).Should(gomega.Succeed())
+
+			//borrowingWL borrows all shared resources and should be scheduled.
+			borrowingWl := testing.MakeWorkload("player2-a", matchingNS.Name).Queue(lqTeamA.
+				Name).Request(corev1.ResourceCPU, "5").Priority(11).Obj()
+			gomega.Expect(k8sClient.Create(ctx, borrowingWl)).Should(gomega.Succeed())
+
+			//blockedWL should try to borrow resources from strictFIFO CQ, but should be blocked
+			//from borrowing because there is a pending workload in strictFIFO CQ.
```

Comment:
```suggestion
			// blockedWL wants to borrow resources from strictFIFO CQ, but should be blocked
			// from borrowing because there is a pending workload in strictFIFO CQ.
```